visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp

```diff
@@ -129,15 +135,28 @@ void MotionPlanningFrame::computePlanButtonClicked()
 void MotionPlanningFrame::computeExecuteButtonClicked()
 {
   if (move_group_ && current_plan_)
-    move_group_->execute(*current_plan_);
+  {
+    ui_->stop_button->setEnabled(true);
+    move_group_->asyncExecute(*current_plan_);
+  }
+}
+
+void MotionPlanningFrame::computeStopButtonClicked()
+{
+  if (move_group_)
+  {
+    move_group_->stop();
+    ui_->result_label->setText("Stopped");
+  }
 }
 
 void MotionPlanningFrame::computePlanAndExecuteButtonClicked()
 {
   if (!move_group_)
     return;
   configureForPlanning();
-  move_group_->move();
+  ui_->stop_button->setEnabled(true);
+  move_group_->asyncMove();
```

Comment:
The button should be only re-enabled after execution. Because you switch to asynchronous execution now, the button gets re-enabled immediately.